.github/workflows/upload-dependencies.yml

```diff
@@ -0,0 +1,67 @@
+---
+## Workflow to push zip with dependencies to S3 bucket every time the ESF version is updated
+name: upload-dependencies
+
+env:
+  BUCKET_NAME : "esf-dependencies"
+  AWS_REGION : "eu-central-1"
+  ROLE: "arn:aws:iam::267093732750:role/esf-dependencies-role"
+
+
+on:
+  workflow_run:
+    workflows: [release]
+    types:
+      - completed
+
+
+permissions:
+  id-token: write   # This is required for requesting the JWT
+  contents: read    # This is required for actions/checkout
+
+
+jobs:
+
+  build-and-upload-dependencies:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      # See https://docs.aws.amazon.com/lambda/latest/dg/python-package.html#python-package-create-dependencies
+
+      - uses: actions/checkout@v4
+
+      - name: Get version number
+        shell: bash
+        run: |
+          VERSION=$(grep -oE '[0-9]+\.[0-9]+\.[0-9]+(\-[a-zA-Z]+[0-9]+)?' share/version.py)
```

Comment:
Perhaps get the tag from git, to avoid duplicating this logic across multiple workflows? e.g. we could use `git describe` to get the tag for the current branch.